openff/nagl_models/_dynamic_fetch.py

```diff
@@ -55,12 +64,33 @@ def get_model(filename: str) -> str:
                 assert cached_path.exists()
                 assert path_to_file == cached_path.as_posix()
 
-                assert _get_sha256(cached_path) == KNOWN_HASHES[filename], (
-                    f"Hash mismatch for {filename}"
-                )
+                if check_hash:
+                    assert (
+                        _get_sha256(cached_path) == check_hash
+                    ), f"Hash mismatch for {filename}"
 
                 return cached_path.as_posix()
 
+    if doi:
+        zenodo_id = re.findall("10.5072/zenodo.([0-9]+)", doi)[0]
```

Comment:
Will we support non-Zenodo DOIs? Figshare etc?

Reply:
Not in [this version](https://github.com/openforcefield/standards/pull/71/files#diff-21013839b35df43712b574d3f5bf302d1f29d6693aaf7e9d3586b98eb89d735aR60) of the NAGLCharges spec, but other sources could be added in a higher NAGLCharges section version.